packages/lesswrong/components/users/DialogueSuggestionsPage.tsx

```diff
@@ -0,0 +1,782 @@
+          to help us know whether the feature is getting used. If one user opts in to revealing their checks we can still not see their matches, unless 
+          the other part of the match has also opted in.</p>      
+        <div className={classes.rootFlex}>
+        <div className={classes.matchContainer}>
+          <h3>Your top upvoted users (last 1.5 years)</h3>
+          <div className={classes.matchContainerGridV1}>
+            <Headers titles={["Dialogue", "Name", "Message", "Match", "Karma", "Agreement", "Posts you've read"]} className={classes.header} />
+            {userDialogueUsefulData.topUsers.slice(0,20).map(targetUser => {
+              const checkId = userDialogueChecks?.find(check => check.targetUserId === targetUser._id)?._id
+              return (
+                <React.Fragment key={targetUser._id}>
+                  <DialogueCheckBox 
+                    targetUserId={targetUser._id}
+                    targetUserDisplayName={targetUser.displayName} 
+                    checkId={checkId} 
+                    isChecked={isChecked(userDialogueChecks, targetUser._id)}
+                    isMatched={isMatched(userDialogueChecks, targetUser._id)}
+                    classes={classes}
+                  />
+                  <UsersName 
+                    className={classes.displayName} 
+                    documentId={targetUser._id} 
+                    simple={false}/>
+                  <MessageButton 
+                    targetUserId={targetUser._id} 
+                    currentUser={currentUser} 
+                    classes={classes} />
+                  <MatchDialogueButton
+                    isMatched={isMatched(userDialogueChecks, targetUser._id)}
+                    targetUserId={targetUser._id}
+                    targetUserDisplayName={targetUser.displayName}
+                    currentUser={currentUser}
+                    loadingNewDialogue={loadingNewDialogue}
+                    createDialogue={createDialogue}
+                    classes={classes}
+                  />
+                  <div className={classes.centeredText}>{targetUser.total_power}</div>
+                  <div className={classes.centeredText}>{targetUser.total_agreement}</div>
+                  <UserPostsYouveRead 
+                    classes={classes} 
+                    targetUserId={targetUser._id} 
+                    components={Components} />
+                </React.Fragment> 
+              )}
+            )}
+          </div>
+        </div>
+      </div>
+      <br />
+      <div className={classes.rootFlex}>
+        <div className={classes.matchContainer}>
+          <h3>Users who published dialogues</h3>
+          <div className={classes.matchContainerGridV2}>
+            <Headers titles={["Dialogue", "Name", "Message", "Match", "Bio", "Posts you've read"]} className={classes.header} />
```

Comment:
I'd extract these to their own variable (I was briefly confused by whether these were the same as in the previous section; having them both be variables with different names would make that obvious to future readers)